fraud/service.go

```diff
@@ -0,0 +1,171 @@
+package fraud
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/celestiaorg/celestia-node/header"
+
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+)
+
+var log = logging.Logger("fraud")
```

Comment:
Also, let's mpve to interfaces.go, to the top

Reply:
done